dashboard/app/jobs.go

```diff
@@ -208,24 +210,38 @@ func saveJob(c context.Context, job *Job, bugKey *db.Key) (*db.Key, error) {
 		CrashReference{CrashReferenceJob, extJobID(jobKey), timeNow(c)})
 }
 
-func checkTestJob(c context.Context, bug *Bug, bugReporting *BugReporting, crash *Crash,
-	repo, branch string) string {
+func patchTestJob(c context.Context, args *testJobArgs) error {
+	if args.branch == "" && args.repo == "" {
+		build, err := loadBuild(c, args.bug.Namespace, args.crash.BuildID)
+		if err != nil {
+			return fmt.Errorf("failed to find the bug reporting object: %w", err)
+		}
+		args.branch = build.KernelBranch
+		args.repo = build.KernelRepo
+	} else if args.branch == "" || args.repo == "" {
+		return &BadTestRequestError{"branch and repo should be either both set or both empty"}
```

Comment:
If input validation is needed, I'd like it to take first function lines.

Reply:
It's actually not necessary -- repo and branch are validated in a separate method. I've dropped the line.